frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -1509,491 +1775,250 @@ function ReaderFooter:addToMainMenu(menu_items)
     if Device:hasBattery() then
-        table.insert(sub_items, getMinibarOption("battery"))
-    end
-    table.insert(sub_items, getMinibarOption("bookmark_count"))
-    table.insert(sub_items, getMinibarOption("percentage"))
-    table.insert(sub_items, getMinibarOption("book_time_to_read"))
-    table.insert(sub_items, getMinibarOption("chapter_time_to_read"))
-    if Device:hasFrontlight() then
-        table.insert(sub_items, getMinibarOption("frontlight"))
+        table.insert(configure_items_sub_table , 5, {
+            text_func = function()
+                if self.settings.battery_hide_threshold <= (Device:hasAuxBattery() and 200 or 100) then
+                    return T(_("Hide battery item when higher than: %1".. "%"), self.settings.battery_hide_threshold)
+                else
+                    return _("Hide battery item at custom threshold")
+                end
+            end,
+            checked_func = function()
+                return self.settings.battery_hide_threshold < MAX_BATTERY_HIDE_THRESHOLD
+            end,
+            enabled_func = function()
+                return self.settings.all_at_once == true
+            end,
+            separator = true,
+            callback = function(touchmenu_instance)
+                local SpinWidget = require("ui/widget/spinwidget")
+                local battery_threshold = SpinWidget:new{
+                    value = math.min(self.settings.battery_hide_threshold, Device:hasAuxBattery() and 200 or 100),
+                    value_min = 0,
+                    value_max = Device:hasAuxBattery() and 200 or 100,
+                    default_value = Device:hasAuxBattery() and 200 or 100,
+                    unit = "%",
+                    value_hold_step = 10,
+                    title_text = _("Minimum threshold to hide battery item"),
+                    callback = function(spin)
+                        self.settings.battery_hide_threshold = spin.value
+                        self:refreshFooter(true, true)
+                        if touchmenu_instance then touchmenu_instance:updateItems() end
+                    end,
+                    extra_text = _("Disable"),
+                    extra_callback = function()
+                        self.settings.battery_hide_threshold = MAX_BATTERY_HIDE_THRESHOLD
+                        if touchmenu_instance then touchmenu_instance:updateItems() end
+                    end,
+                    ok_always_enabled = true,
+                }
+                UIManager:show(battery_threshold)
+            end,
+            keep_menu_open = true,
+        })
     end
-    if Device:hasNaturalLight() then
-        table.insert(sub_items, getMinibarOption("frontlight_warmth"))
+    -- More status bar options
+    -- quick access to this setting for "@NiLuJe, and for people that do like him." -- @poire-z (2024)
+    table.insert(sub_items, getMinibarOption("reclaim_height"))
```

Comment:
I think this one deserves a help_text.
These "options" can have a help_text, look for `option_help_text`.

Reply:
tried adding, 

```lua
option_help_text[MODE.reclaim_height] = _("When status bar is unlocked and hidden, this setting will utilise the entirety of screen real state and will temporarily overlap status bar and text when unhidden.")
``` 

to the existing array and it crashes the program complaining about a loop...

Reply:
Side effect of the accumulated crap that readerfooter has become, with eveyrbody hacking its own little quick ad hoc solution to what one wanted at the time, and there's many things inconsistent...
There's no real reason to use MODE.xyz or MODE[option] in these bits:
https://github.com/koreader/koreader/blob/2a49d5c9458c9a070599d00511f3a5a7da2de336/frontend/apps/reader/modules/readerfooter.lua#L678-L683
https://github.com/koreader/koreader/blob/2a49d5c9458c9a070599d00511f3a5a7da2de336/frontend/apps/reader/modules/readerfooter.lua#L1021-L1026

(Mode is an integer from https://github.com/koreader/koreader/blob/2a49d5c9458c9a070599d00511f3a5a7da2de336/frontend/apps/reader/modules/readerfooter.lua#L32-L51 , reclaim_height is not part of them.

Just use the strings and drop the `MODE.` / `MODE[]` indirection:
```lua
option_help_text["percentage"] = _("Progress percentage ...")
[...]
 help_text = type(option_help_text[option]) == "string" 
     and option_help_text[option], 
 help_text_func = type(option_help_text[option]) == "function" and 
     function(touchmenu_instance) 
         option_help_text[option](self, touchmenu_instance) 
     end, 
```

Reply:
cheers, I can't test that is working (those long-press-help-text messages can't be triggered on non-touch), but at least is not crashing anymore.